src/sql/workbench/contrib/notebook/browser/models/untitledNotebookInput.ts

```diff
@@ -26,6 +26,8 @@ export class UntitledNotebookInput extends NotebookInput {
 		@IExtensionService extensionService: IExtensionService
 	) {
 		super(title, resource, textInput, true, textModelService, instantiationService, notebookService, extensionService);
+		// Set the mode explicitly so that the auto language detection doesn't run and mark the model as being JSON
+		this.textInput?.resolve().then(() => this.setMode('notebook'));
```

Comment:
Don't fix test errors like this - you should fix the test to pass in a dummy (test) input instead.

Reply:
why would I fix the test and not the root cause? What if there's a scenario where textInput isn't defined and resolve would fail ?

Reply:
Because it's not a valid scenario - if a Notebook doesn't have a text input then something has gone horribly wrong (every input needs to have a backing model or it doesn't make sense)